plsql/ExcelGen.pkb

```diff
@@ -1757,7 +1761,9 @@ create or replace package body ExcelGen is
             else
               data.varchar2_value := to_char(data.number_value, 'TM9', NLS_PARAM_STRING);
             end if;
-            stream_write(stream, '<c r="'||cellRef||'"><v>'||data.varchar2_value||'</v></c>');
+            stream_write(stream, '<c r="'||cellRef
+                ||case when numXfId is not null then '" s="'||to_char(numXfId) end
```

Comment:
The condition must be `numXfId != 0` here.

Reply:
Apologies, but I do not understand the "styles" model.  I simply copied the code for dates. I now understand it will not be null. I'm not following how putCellXf will be returning 0 when passed a null varchar2 for number format. There must be some magic I missed that creates an empty style in the first position. I'll take your word for it.

Reply:
Yes, the styles model is a little complicated IMO. 
Basically, an XfId is a 0-based index of a style definition in the collection of styles we may apply to a cell.
Each style definition is composed of attributes referencing number format, font, fill, and border through indices into their respective collections as well. That way, a given style feature (e.g. a font definition) is stored only once and may be referenced multiple times.

The `newStylesheet` function (called by `new_workbook`) takes care of creating the default (empty) style at index 0.